src/racer/core.rs

```diff
@@ -286,14 +286,15 @@ impl fmt::Debug for PathSearch {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "Search [{:?}, {:?}, {:?}]",
                self.path,
-               self.filepath.to_str(),
+               self.filepath.display(),
                self.point)
     }
 }
 
 pub struct IndexedSource {
     pub code: String,
-    pub idx: Vec<(usize, usize)>
+    pub idx: Vec<(usize, usize)>,
+    pub lines: RefCell<Vec<(usize, usize)>>
```

Comment:
I wanted to initialize the indices lazily because they're probably not used for many files.

But it could conceivably be done in the same iteration that generates the statement indices.